### ISSUE TITLE
trim incorrect result in case of BOTH

### ISSUE THREAD

**Issue by den-crane** [ClickHouse/ClickHouse#30245]
```sql
SELECT trim(BOTH ', ' FROM '5935,5998,6014, ')||'|' x

SELECT concat(replaceRegexpAll('5935,5998,6014, ', concat('^[', regexpQuoteMeta(', '), ']*|[', regexpQuoteMeta(', '), ']*$'), ''), '|') AS x

┌─x─────────────────┐
│ 5935,5998,6014, | │
└───────────────────┘

psql> SELECT trim(BOTH ', ' FROM '5935,5998,6014, ')||'|'
5935,5998,6014|

```
http://sqlfiddle.com/#!17/0a28f/406

though trailing
```sql
SELECT trim(trailing ', ' FROM '5935,5998,6014, ')||'|' x
┌─x───────────────┐
│ 5935,5998,6014| │
└─────────────────┘
```

**Comment by Joeywzr** [comment]
Let me have a try

**Comment by den-crane** [comment]
```sql
select replaceRegexpAll(',,1,,','^[,]*|[,]*$', '') x;
┌─x─┐
│ 1 │
└───┘

select replaceRegexpAll(',,1','^[,]*|[,]*$', '') x;
┌─x─┐
│ 1 │
└───┘

select replaceRegexpAll('1,,','^[,]*|[,]*$', '') x;
┌─x───┐
│ 1,, │
└─────┘

```

**Comment by alexey-milovidov** [comment]
The fix #30292 was wrong.
